src/city_score/sources/traillink.py

```diff
@@ -0,0 +1,87 @@
+from ..cache import cache
+from ..decorators import dimension, scorer
+from ..source import Source
+from html.parser import HTMLParser
+import requests
+
+activities_filter = {
+    'Atv': 'ATV',
+    'Bike': 'BIKE',
+    'Birding': 'BIRD',
+    'Cross Country Skiing': 'XSKI',
+    'Dog Walking': 'DOG',
+    'Fishing': 'FISH',
+    'Geocaching': 'GEO',
+    'Hiking': 'HIKE',
+    'Horseback Riding': 'HORSE',
+    'Inline Skating': 'SKTS',
+    'Mountain Biking': 'MTBK',
+    'Running': 'RUN',
+    'Snowmobiling': 'SNOW',
+    'Walking': 'WALK',
+    'Wheelchair Accessible': 'WHEEL'
+}
+
+class TrailLinkMilesParser(HTMLParser):
+    def __init__(self):
+        super().__init__()
+        self.found = False
+        self.total_trail_miles = 0
+
+    def handle_starttag(self, tag, attrs):
+        if tag == 'span':
+            for attr in attrs:
+                if attr[0] == 'itemprop' and attr[1] == 'distance':
+                    self.found = True
+
+    def handle_data(self, data):
+        if self.found:
+            self.total_trail_miles += float(data.split()[0])
+
+    def handle_endtag(self, tag):
+        if tag == 'span' and self.found:
+            self.found = False
+
+class TrailLink(Source):
+    base_url = 'https://www.traillink.com/trailsearch'
+
+def format(number):
+    return f'{number} miles'
+
+@dimension('TrailLink Mileage')
+def trail_total_miles(city, activities=[], min_length=5):
+    key = 'traillink-%s-%s' % (str(city), str("-".join(activity.replace(" ", "") for activity in activities)))
+    count = cache.get(key)
+    if count is not None:
+        return format(count)
+    
+    activity_values = []
+    for activity in activities:
+        activity_values.append(activities_filter[activity])
+
+    params={
+        'city': city.name,
+        'state': city.state,
+        'activities': ",".join(activity_values),
+        'length': f'{min_length}|99999'
+    }
+    response = requests.get(TrailLink.base_url, params)
+
+    if response.status_code == 200:
+        parser = TrailLinkMilesParser()
+        parser.feed(response.text)
+        count = int(parser.total_trail_miles)
+        cache.set(key, count)
+        return format(count)
+
+@scorer('TrailLink')
+def trail_total_miles_scorer(city, lower, upper, activities=[]):
```

Comment:
@jacobbudin a little confused about what values `lower` / `upper` represent here? i dont know what the max value of trail mileage would be across all cities

Reply:
ah I see. the `upper` then is sort of an arbitrary value the user chooses? wonder if we could grab the max value out of the cities and then use that as upper to scale everything else

Reply:
@grelas Yes, that'd be a great addition. There should be a "full auto" mode where it plucks all the minimums and maximums and uses those as the lower and upper bounds for users who prefer a quick initial result without investing much time in tuning.

That said, `upper` and `lower` is an intentional design decision:
- e.g., a `lower=60` on life expectancy in years, if people are dying before age 60, it's a crummy town and should get zero points 😄 
- e.g., a `upper=10` on number of bookstores or movie theaters or Vietnamese restaurants, because the eleventh extra business is not going to materially increase your satisfaction with a city, ten is probably "enough"

I'd also like to explore non-linear scorers ([logarithmic?](https://saylordotorg.github.io/text_intermediate-algebra/s10-03-logarithmic-functions-and-thei.html)). Feel free to submit that PR!